java/ql/src/semmle/code/java/security/SecurityFlag.qll

```diff
@@ -0,0 +1,84 @@
+/**
+ * Provides utility predicates to spot variable names, parameter names, and string literals that suggest deliberately insecure settings.
+ */
+
+import java
+import semmle.code.java.controlflow.Guards
+import semmle.code.java.dataflow.DataFlow
+import semmle.code.java.dataflow.FlowSources
+
+/**
+ * A kind of flag that may indicate security expectations regarding the code it guards.
+ */
+abstract class FlagKind extends string {
+  bindingset[this]
+  FlagKind() { any() }
+
+  /**
+   * Returns a flag name of this type.
+   */
+  bindingset[result]
+  abstract string getAFlagName();
+
+  /** Gets a node representing a (likely) security flag. */
+  DataFlow::Node getAFlag() {
+    exists(VarAccess v | v.getVariable().getName() = getAFlagName() |
+      result.asExpr() = v and v.getType() instanceof FlagType
+    )
+    or
+    exists(StringLiteral s | s.getRepresentedString() = getAFlagName() | result.asExpr() = s)
+    or
+    exists(MethodAccess ma | ma.getMethod().getName() = getAFlagName() |
+      result.asExpr() = ma and
+      ma.getType() instanceof FlagType
+    )
+    or
+    flagFlowStep*(getAFlag(), result)
+  }
+}
+
+/**
+ * Flags suggesting an optional feature, perhaps deliberately insecure.
+ */
+class SecurityFeatureFlag extends FlagKind {
```

Comment:
```suggestion
private class SecurityFeatureFlag extends FlagKind {
```

Reply:
This doesn't need to be public, does it?

Reply:
Sorry, I missed that, you're right.